crates/astria-sequencer/src/transaction/mod.rs

```diff
@@ -84,26 +86,37 @@ impl std::error::Error for InvalidNonce {}
 
 #[async_trait::async_trait]
 impl ActionHandler for UnsignedTransaction {
-    fn check_stateless(&self) -> anyhow::Result<()> {
+    async fn check_stateless(&self) -> anyhow::Result<()> {
         ensure!(!self.actions.is_empty(), "must have at least one action");
 
         for action in &self.actions {
             match action {
                 Action::Transfer(act) => act
                     .check_stateless()
+                    .await
                     .context("stateless check failed for TransferAction")?,
                 Action::Sequence(act) => act
                     .check_stateless()
+                    .await
                     .context("stateless check failed for SequenceAction")?,
                 Action::ValidatorUpdate(act) => act
                     .check_stateless()
+                    .await
                     .context("stateless check failed for ValidatorUpdateAction")?,
                 Action::SudoAddressChange(act) => act
                     .check_stateless()
+                    .await
                     .context("stateless check failed for SudoAddressChangeAction")?,
+                Action::Ibc(act) => {
+                    act.clone()
+                        .with_handler::<crate::accounts::ics20_transfer::Ics20Transfer>()
+                        .check_stateless(())
```

Comment:
That's their (penumbra's) `check_stateless`, right?

Yeah, I see the `penumbra` action handler above. can we make this more obvious to the reader to disambiguate between our `check_stateless` and theirs?


```rust
let the_action = act.clone().with_handler::<Ics20Transfer>();
penumbra_component::ActionHandler::check_stateless(&the_action);
```